rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,205 @@
+---
+
+# Summary
+[summary]: #summary
+
+Stabilize the new command line interface and Flakes in an incremental fashion, leveraging [RFC 134](https://github.com/NixOS/rfcs/pull/134).
+This plan is designed to still efficiently end the current "limbo" era of these unstable yet widely-used features, yet also soothe the longstanding tensions in the Nix community over how we got here.
+
+# Motivation
+[motivation]: #motivation
+
+For the past few years, we've accumulated a massive backlog of unstable features in the form of the new command-line interface and flakes.
+There is now a growing desire to stabilize those features, but as of yet no plan on exactly how to do so.
+The tension between that desire and a lack of a clear plan has loomed over our heads for a while; this RFC aims to dispel it by providing a concrete plan, an plan that hopefully will mitigate the lingering controversies and tensions around Flakes.
+
+## The situation so far
+
+There are a few facts that must be introduced to give context to this situation.
+We can't change these short of a time machine, so we must find a way to live with them instead.
+
+### Flakes are very popular
+
+As measured in the community polls, Nix has a lot of new users, dwarfing the number of long-time users.
+Flakes is very popular among these new users.
+Ergo, Flakes is very popular among the Nix community as a whole.
+
+Like it or not, these users have been using Flakes as if it was stable, and we cannot make huge drastic changes that would break their code in hard-to-fix ways.
+
+Many Groups and individuals interested in the continued growth the Nix community see Flakes are popular, an also wish it to be stabilized to attract further still more users, since Flakes are already proven to be popular among users.
+
+### Difficulties in the roll-out
+
+It is undeniable that the roll-out of these new features has been controversial.
+Graham Christensen's blog post [flakes-are-an-obviously-good-thing](https://grahamc.com/blog/flakes-are-an-obviously-good-thing) lays out the procedure issues along the way quite well.
+
+Some people were upset the Flakes RFC was abandoned but the feature merged.
+Other people were fine with experimental features being merged without RFC, but were upset because Flakes never really *felt* experimental.
+Experimental features would be modified a lot subject to community design use, and, most importantly, *discouraged* from being used in production.
+
+### Flakes are criticized for encroaching on other features
+
+There are many criticism about Flakes.
+But one of them especially relevant to stabilizing is a perception that Flakes have encroached on other new features, in the sense that it ought to be possible to use them without Flakes but isn't in practice.
+For example, there is no reason in theory pure eval requires Flakes, but without the ability to populate an initial whitelist of store paths that are safe to import it in practice does.
```

Comment:
```suggestion
For example, there is no reason in theory that pure evaluation if Nix expressions requires Flakes. But without the ability to populate an initial allow list of store paths that are safe to import, it in practice does.
```

Maybe link to the manual on that setting for people to find what that means?

Does it though? Why can't pure eval constrain paths to descendants of the CWD or something? It appears to me these considerations simply haven't been written down so I still have to ask questions that are probably stupid and have been answered multiple times over.

Reply:
```suggestion
For example, there is no reason in theory that pure evaluation if Nix expressions requires Flakes.
But without the ability to populate some sort of initial list of store paths that are safe to import, pure evaluation in practice does require Flakes.
```